api/src/main/java/com/velocitypowered/api/proxy/Player.java

```diff
@@ -275,4 +282,27 @@ default void sendMessage(net.kyori.text.Component component) {
    */
   @Override
   boolean sendPluginMessage(ChannelIdentifier identifier, byte[] data);
+
+  @Override
+  default @NotNull Key key() {
+    return Key.key("player");
+  }
+
+  /**
+   * Gets a {@link Component} that renders a player name similarly to vanilla.
+   *
+   * @return a Component representing this player
+   */
+  default @NotNull Component getDisplayName() {
```

Comment:
There have been prior discussions on including a display name component API in Velocity, but this isn't the way to go about it. In some cases, this is even entirely inappropriate because `/tell` might not exist or the user doesn't want to provide an insertion. The only sane implementation of this is `Component.text(getUsername())` and have this API hidden in the pointer, unless your plan is to add in the display name component API (which would have to be done for a future version of Velocity).

Reply:
I'd prefer we keep display name stuff out of this pull request, personally.

Reply:
display name stuff has been removed 👍🏻